Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsConnectorAzureFunction/DS_poller.py

```diff
@@ -0,0 +1,155 @@
+                    logger.info("Total number of events are " + str(len(event_data)))
+                else:
+                    logger.info("There are no new events")
+                    return [], int
+
+            
+            for event in event_data:
+                if(event is not None and event['triage-item-id'] not in triage_id):
+                    triage_id.append(event['triage-item-id'])
+
+            logger.info(triage_id)
+        except Exception:            
+            logger.exception("Error while getting triage data: ")
+        
+        item_data = self.DS_obj.get_triage_items(triage_id)
+        
+        return item_data, max_event_num
+
+    def poll(self):
+        """
+            main polling function, 
+            makes api calls in following fashion:
+            triage-events --> triage-items --> incidents and alerts 
+        """
+        item_data = []            
+        try:
+            #sending data to sentinel
+            inc_ids = []
+            alert_ids = []
+            item_data, max_event_num = self.get_data()
+            if(item_data):
+                logger.info("total number of items are " + str(len(item_data)))
+                #creating list of ids by alert and incidents
+                alert_triage_items = list(filter(lambda item: item['source']['alert-id'] is not None, item_data))
+                inc_triage_items = list(filter(lambda item: item['source']['incident-id'] is not None, item_data))
+                
+                #getting data from DS and posting to Sentinel
+                if inc_triage_items:
+                    inc_ids = [item['source']['incident-id'] for item in inc_triage_items]
+                    response_inc = self.DS_obj.get_incidents(inc_ids)
+                    if(not 200 <= int(response_inc.status_code) < 300):
+                        raise RuntimeError("Unexpected response from portal while getting incidents. Status code: {}".format(response_inc.status_code))        
+                        
+                if alert_triage_items:
+                    alert_ids = [item['source']['alert-id'] for item in alert_triage_items]
+                    response_alert = self.DS_obj.get_alerts(alert_ids)
+                    if(not 200 <= int(response_alert.status_code) < 300):
+                        raise RuntimeError("Unexpected response from portal while getting alerts. Status code: {}".format(response_alert.status_code))    
+                    
+                if inc_triage_items:
+                    self.post_azure(response_inc, inc_triage_items)
+                if alert_triage_items:
+                    self.post_azure(response_alert, alert_triage_items)
+
+                #saving event num for next invocation
+                self.date.post_event(max_event_num)
+        except Exception:
+            if(item_data):
+                logger.exception("Error polling: ")
+            else:
+                logger.info("No new events found.")
```

Comment:
this log statement belongs in an `else` block alongside `if item_data`, not in a catch block.